### PR TITLE
mp-4752 remove header removes all headers

### PR DESCRIPTION
**Pull request by elan-sg**

### CODE REVIEW

File: gmime/envelope.go [hunk 1]
```diff
@@ -82,6 +82,19 @@ func (m *Envelope) RemoveHeader(name string) bool {
 	return gobool(C.g_mime_header_list_remove(headers, C.CString(name)))
 }
 
+// RemoveAllHeaders removes all headers with the name if there are multiple
+func (m *Envelope) RemoveAllHeaders(name string) bool {
+	headers := C.g_mime_object_get_header_list(m.asGMimeObject())
+	maybeMore := true
+	removed := false
+	for maybeMore {
+		maybeMore = C.g_mime_header_list_remove(headers, C.CString(name)) == 1
```
Comment (Dharmik-Patel) [inline gmime/envelope.go]: do you not have to convert this to gobool?

Comment (elan-sg) [reply gmime/envelope.go]: no its already go

File: gmime/envelope.go [hunk 2]
```diff
@@ -82,6 +82,19 @@ func (m *Envelope) RemoveHeader(name string) bool {
 	return gobool(C.g_mime_header_list_remove(headers, C.CString(name)))
 }
 
+// RemoveAllHeaders removes all headers with the name if there are multiple
+func (m *Envelope) RemoveAllHeaders(name string) bool {
+	headers := C.g_mime_object_get_header_list(m.asGMimeObject())
+	maybeMore := true
```
Comment (Dharmik-Patel) [inline gmime/envelope.go]: can we name it something like hasNext?

Comment (elan-sg) [reply gmime/envelope.go]: that would make the initialization untrue

File: gmime/envelope.go (hunk 1 above)
Comment (Dharmik-Patel) [reply gmime/envelope.go]: but its making call to c lib

**Review by Dharmik-Patel — APPROVED** [review]

**Pull request merged**